### PR TITLE
[AutoML] Add AutoML XML documentation to all public members; migrate AutoML projects & tests into ML.NET solution; AutoML test fixes

### PR DESCRIPTION
**Pull request by daholste**

### CODE REVIEW

File: src/Microsoft.ML.Auto/API/AutoCatalog.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        {
+            return new MulticlassClassificationExperiment(_context, experimentSettings);
+        }
+
+        /// <summary>
+        /// Infers information about the columns of a dataset in a file located at <paramref name="path"/>.
+        /// </summary>
+        /// <param name="path">Path to a dataset file.</param>
+        /// <param name="labelColumnName">The name of the label column.</param>
+        /// <param name="separatorChar">The character used as separator between data points in a row. If <see langword="null"/>, AutoML will try to infer this value.</param>
```
Comment (justinormont) [inline src/Microsoft.ML.Auto/API/AutoCatalog.cs]: I think of an entire row as (potentially) a singular multi-dimensional "data point", perhaps "data element" is better?
```suggestion
        /// <param name="separatorChar">The character used as separator between data elements in a row. If <see langword="null"/>, AutoML will try to infer this value.</param>
```

Comment (daholste) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: this is much better -- thank you!

File: src/Microsoft.ML.Auto/API/AutoCatalog.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        /// <param name="allowQuoting">Whether the file can contain column defined by a quoted string. If <see langword="null"/>, AutoML will try to infer this value.</param>
```
Comment (justinormont) [inline src/Microsoft.ML.Auto/API/AutoCatalog.cs]:
```suggestion
        /// <param name="allowQuoting">Whether the file can contain columns defined by a quoted string. If <see langword="null"/>, AutoML will try to infer this value.</param>
```

Comment (daholste) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: thank you!

Comment (daholste) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: this was a copy-paste error from https://github.com/dotnet/machinelearning/blob/70ef7ecd43b031b481a4047ea361da5e2f360336/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs#L22

Comment (daholste) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: will make a Github issue

File: src/Microsoft.ML.Auto/API/AutoCatalog.cs (hunk 1 above)
Comment (justinormont) [inline src/Microsoft.ML.Auto/API/AutoCatalog.cs]: We likely should move to `char[] Separators` to match ML.NET.

Ref: https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.textloader.options.separators?view=ml-dotnet

File: src/Microsoft.ML.Auto/API/AutoCatalog.cs (hunk 2 above)
Comment (daholste) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: https://github.com/dotnet/machinelearning/pull/3355/files

File: src/Microsoft.ML.Auto/API/AutoCatalog.cs (hunk 1 above)
Comment (daholste) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: Agreed that this warrants further discussion --
I believe we went this direction b/c ML.NET's analog method takes a single separator character as well --https://github.com/dotnet/machinelearning/blob/70ef7ecd43b031b481a4047ea361da5e2f360336/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs#L27

File: src/Microsoft.ML.Auto/API/ExperimentSettings.cs [hunk 3]
```diff
@@ -23,10 +51,10 @@ public class ExperimentSettings
         /// <summary>
         /// This setting controls whether or not an AutoML experiment will make use of ML.NET-provided caching.
         /// If set to true, caching will be forced on for all pipelines. If set to false, caching will be forced off.
-        /// If set to null (default value), AutoML will decide whether to enable caching for each model.
+        /// If set to <see langword="null"/> (default value), AutoML will decide whether to enable caching for each model.
         /// </summary>
         public bool? CacheBeforeTrainer = null;
```
Comment (justinormont) [inline src/Microsoft.ML.Auto/API/ExperimentSettings.cs]: We had a quorum of folks decide this should be an enum: on/off/auto. Welcome to add into this PR; or we'll handle later.

File: src/Microsoft.ML.Auto/API/AutoCatalog.cs (hunk 1 above)
Comment (justinormont) [reply src/Microsoft.ML.Auto/API/AutoCatalog.cs]: Ok. Sounds good.

File: src/Microsoft.ML.Auto/API/ExperimentSettings.cs (hunk 3 above)
Comment (daholste) [reply src/Microsoft.ML.Auto/API/ExperimentSettings.cs]: +1 -- will handle in a PR shortly w/ a few other API revs

**Pull request merged**